src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,159 @@
+import React, {
+  useCallback, useEffect, useRef, useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import './TodoInfo.scss';
+
+interface TodoInfoProps {
+  todo: Todo;
+  onRemoveTodo: (id: number) => void;
+  onUpdateTodo: (todoId: number, updatedValue: string | boolean) => void;
+  loadingTodoIds: number[];
+}
+
+export const TodoInfo: React.FC<TodoInfoProps> = ({
+  todo,
+  onRemoveTodo,
+  onUpdateTodo,
+  loadingTodoIds,
+}) => {
+  const { id, title, completed } = todo;
+
+  const [completedStatus, setCompletedStatus] = useState(completed);
+  const [isActiveEditForm, setIsActiveEditForm] = useState(false);
+  const [currentTitle, setCurrentTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isActiveEditForm && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isActiveEditForm]);
+
+  const updateTitle = useCallback(() => {
+    const formattedTitle = currentTitle
+      .replace(/\s{2,}/, ' ')
+      .trim();
+
+    if (formattedTitle === title) {
+      setIsActiveEditForm(false);
+
+      return;
+    }
+
+    if (!formattedTitle) {
+      onRemoveTodo(id);
+
+      return;
+    }
+
+    onUpdateTodo(id, formattedTitle);
+
+    setCurrentTitle(title);
+  }, [currentTitle]);
+
+  const handleStatusChange = () => {
+    const updatedCompletedStatus = !completed;
+
+    setCompletedStatus(updatedCompletedStatus);
+
+    onUpdateTodo(id, updatedCompletedStatus);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentTitle(event.target.value);
+  };
+
+  const handleRemoveClick = () => {
+    onRemoveTodo(id);
+  };
+
+  const handleDoubleClick = () => {
+    const updatedStatus = !isActiveEditForm;
+
+    setIsActiveEditForm(updatedStatus);
+  };
+
+  const handleKeyTouch = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.code === 'Escape') {
```

Comment:
It would be very cool to put such strings into separate variables to avoid using magic data.
For example:
```
const ESCAPE= 'Escape';
const NOT_SET_VALUE = '-';
```
This applies to all code.